src/query/functions/tests/it/type_check/testdata/decimal.txt

```diff
@@ -0,0 +1,4096 @@
++--------+-----------------------+
+| d128   | Decimal128([0])       |
+| Output | Boolean([0b_______0]) |
++--------+-----------------------+
+
+
+ast            : d128 = 1.0
+raw expr       : eq(d128::Decimal(38, 0), 1.0)
+checked expr   : eq<Decimal(38, 0), Decimal(2, 1)>(d128, 1.0_d128(2,1))
+optimized expr : false
+evaluation:
++--------+----------------+---------+
+|        | d128           | Output  |
++--------+----------------+---------+
+| Type   | Decimal(38, 0) | Boolean |
+| Domain | {0..=0}        | {FALSE} |
+| Row 0  | 0              | false   |
++--------+----------------+---------+
+evaluation (internal):
++--------+-----------------------+
+| Column | Data                  |
++--------+-----------------------+
+| d128   | Decimal128([0])       |
+| Output | Boolean([0b_______0]) |
++--------+-----------------------+
+
+
+ast            : 1.0 > d128
+raw expr       : gt(1.0, d128::Decimal(38, 0))
+checked expr   : gt<Decimal(2, 1), Decimal(38, 0)>(1.0_d128(2,1), d128)
+optimized expr : true
+evaluation:
++--------+----------------+---------+
+|        | d128           | Output  |
++--------+----------------+---------+
+| Type   | Decimal(38, 0) | Boolean |
+| Domain | {0..=0}        | {TRUE}  |
+| Row 0  | 0              | true    |
++--------+----------------+---------+
+evaluation (internal):
++--------+-----------------------+
+| Column | Data                  |
++--------+-----------------------+
+| d128   | Decimal128([0])       |
+| Output | Boolean([0b_______1]) |
++--------+-----------------------+
+
+
+ast            : 1.0 = d128
+raw expr       : eq(1.0, d128::Decimal(38, 0))
+checked expr   : eq<Decimal(2, 1), Decimal(38, 0)>(1.0_d128(2,1), d128)
+optimized expr : false
+evaluation:
++--------+----------------+---------+
+|        | d128           | Output  |
++--------+----------------+---------+
+| Type   | Decimal(38, 0) | Boolean |
+| Domain | {0..=0}        | {FALSE} |
+| Row 0  | 0              | false   |
++--------+----------------+---------+
+evaluation (internal):
++--------+-----------------------+
+| Column | Data                  |
++--------+-----------------------+
+| d128   | Decimal128([0])       |
+| Output | Boolean([0b_______0]) |
++--------+-----------------------+
+
+
+error: 
+  --> SQL:1:6
+  |
+1 | d128 > '1'
+  |      ^ no function matches signature `gt(Decimal(38, 0), String)`, you might need to add explicit type casts.
+
+candidate functions:
+  gt(Variant, Variant) :: Boolean                               : unable to unify `Decimal(38, 0)` with `Variant`
+  gt(Variant NULL, Variant NULL) :: Boolean NULL                : unable to unify `Decimal(38, 0)` with `Variant`
+  gt(String, String) :: Boolean                                 : unable to unify `Decimal(38, 0)` with `String`
+... and 33 more
+
+
+
+ast            : d128 = '1'
+raw expr       : eq(d128::Decimal(38, 0), '1')
+checked expr   : eq<Float64, Float64>(to_float64<Decimal(38, 0)>(d128), to_float64<String>("1"))
```

Comment:
The implicit type conversion rules are a bit inconsistent here. 
Why `d128 = '1'` is allowed but `d128 > '1'` is not. and should not cast to Float64